### DISCUSSION TITLE
How to handle balance fetching with wallet connect v2?

### DISCUSSION THREAD

**Discussion by baymac** [WalletConnect/walletconnect-monorepo#2726] [Developer Support]
For signing transaction requests there will be a pop or push notification by sign client when the wallet app is cleared from memory. But `eth_getBalance` request doesn't work when the wallet is cleared from memory. 

Which of the following 2 methods to get balance is correct:

1. Using `eth_getBalance` RPC call to wallet (this doesn't work when the wallet is cleared from memory in user's device)
2. Using some indexer like infura

**Comment by glitch-txs** [comment]
what do you mean by 'cleared from memory'?

**Reply by baymac** [reply]
Removed from recents (to be more clear, like in this video https://docs.walletconnect.com/assets/sign-notifications-android.mov)
